modules/cudaoptflow/CMakeLists.txt

```diff
@@ -6,4 +6,23 @@ set(the_description "CUDA-accelerated Optical Flow")
 
 ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4127 /wd4324 /wd4512 -Wundef -Wmissing-declarations -Wshadow)
 
+# ----------------------------------------------------------------------------
+# MODULE REQUIREMENTS
+# ----------------------------------------------------------------------------
+
+set(NVIDIA_OPTICAL_FLOW_1_0_HEADERS_PATH "${CMAKE_CURRENT_SOURCE_DIR}/include/opencv2/NVIDIAOpticalFlowSDK_1_0_Headers")
+set(NVIDIA_OPTICAL_FLOW_1_0_HEADERS_ROOT "${NVIDIA_OPTICAL_FLOW_1_0_HEADERS_PATH}/NVIDIAOpticalFlowSDK_1_0_Headers")
+ocv_download(FILENAME "NVIDIAOpticalFlowSDK_1_0_Headers.tar.gz"
+               HASH "c9e06b032b9839e7106baf42cb1b429c"
+               URL
+                 "https://raw.githubusercontent.com/NVIDIA/NVIDIAOpticalFlowSDK/nvof_1_0_bsd/"
```

Comment:
BTW, It is possible to download using specific commit hash (no need for archive in git repo):
https://github.com/NVIDIA/NVIDIAOpticalFlowSDK/archive/79c6cee80a2df9a196f20afd6b598a9810964c32.zip

(these files are usually stable, but may change if GitHub changes packing options - I remember only one case when archive's contents are changed (due compression parameters)).

Reply:
Oh Thanks. I was not aware about it.
Il make it download via hash.

Reply:
Any suggestions on this @alalek

Reply:
> d41d8cd98f00b204e9800998ecf8427e

Why did you using MD5 of empty file?
Take a look on CMake logs, error message is there.

It should be:
```
set(NVIDIA_OPTICAL_FLOW_1_0_HEADERS_MD5 "ca5acedee6cb45d0ec610a6732de5c15")
```

BTW, files are unpacked with extra sub-directory:
```
include/opencv2/NVIDIAOpticalFlowSDK_1_0_Headers/
└── NVIDIAOpticalFlowSDK-79c6cee80a2df9a196f20afd6b598a9810964c32
    ├── nvOpticalFlowCommon.h
    ├── nvOpticalFlowCuda.h
    └── README.md
```

So, we need:
```
ocv_include_directories(SYSTEM "${NVIDIA_OPTICAL_FLOW_1_0_HEADERS_PATH}/NVIDIAOpticalFlowSDK-${NVIDIA_OPTICAL_FLOW_1_0_HEADERS_MD5}")
```

and remove these headers from public API location:

```
set(NVIDIA_OPTICAL_FLOW_1_0_HEADERS_PATH "${CMAKE_CURRENT_BINARY_DIR}/include/NVIDIAOpticalFlowSDK_1_0_Headers")
```

Reply:
Done